extensions/default/src/MergeDataSource/index.js

```diff
@@ -0,0 +1,195 @@
+      level,
+    });
+    data = data.concat(dedupedData);
+  } else {
+    data = data.concat(incomingData);
+  }
+  globalData.push(tagFunc(data, sourceName));
+};
+
+function createMergeDataSourceApi(mergeConfig, UserAuthenticationService, extensionManager) {
+  const { seriesMerge } = mergeConfig;
+  const { dataSourceNames } = seriesMerge;
+
+  /**
+   * Calls a specified function on all data sources and merges the results.
+   *
+   * @async
+   * @function
+   * @param {string} path - The path of the function to call.
+   * @param {Object} args - The arguments to pass to the function.
+   * @returns {Promise<Array>} - A promise that resolves to an array of merged data.
+   */
+  const callForAllDataSourcesAsync = async (path, args) => {
+    const mergeMap = {
+      'query.studies.search': {
+        mergeKey: 'studyInstanceUid',
+        level: 0,
+        /** TODO */
+        tagFunc: x => x,
+      },
+      'retrieve.series.metadata': {
+        mergeKey: 'seriesInstanceUid',
+        level: 0,
+        tagFunc: (series, sourceName) => {
+          series.forEach(series => {
+            series.RetrieveAETitle = sourceName;
+            DicomMetadataStore.updateSeriesMetadata(series);
+          });
+          return series;
+        },
+      },
+    };
+
+    const { mergeKey, level, tagFunc } = mergeMap[path] || { tagFunc: x => x, level: -1 };
+
+    const dataSourceDefs = Object.values(extensionManager.dataSourceDefs);
+    const promises = [];
+    let globalData = [];
+
+    for (const dataSourceDef of dataSourceDefs) {
+      const { configuration, sourceName } = dataSourceDef;
+      if (configuration && dataSourceNames.includes(sourceName)) {
+        const [dataSource] = extensionManager.getDataSources(sourceName);
+        const promise = executeFunction(dataSource, path, args);
+        promises.push(promise);
+
+        promise.then(data =>
+          processDedupedData({
+            sourceName,
+            globalData,
+            incomingData: data,
+            mergeKey,
+            level,
+            tagFunc,
+          })
+        );
+      }
+    }
+
+    await Promise.allSettled(promises);
+
+    return globalData.flat();
+  };
+
+  /**
+   * Calls a function with the given path and arguments on all data sources that are included in the given list of data source names.
+   * Returns an array of the results from each data source.
+   *
+   * @param {string} path - The path of the function to call on each data source.
+   * @param {Array} args - The arguments to pass to the function on each data source.
+   * @returns {Array} - An array of the results from each data source.
+   */
+  const callForAllDataSources = (path, args) => {
+    const dataSourceDefs = Object.values(extensionManager.dataSourceDefs);
+    const globalData = [];
+    for (const dataSourceDef of dataSourceDefs) {
+      const { configuration, sourceName } = dataSourceDef;
+      if (configuration && dataSourceNames.includes(sourceName)) {
```

Comment:
This should be === as I think the includes sourceName is tricky to ensure you get it right.  
If you want a way to group calls, then add a parameter to the configuration that allows you to compare for exact equality on that parameter value.  You could make that parameter keyed so that someone could create a new/arbitrary key for this.

Reply:
The includes here is running against an array, and in that case, it's the same as ===. You can test it by running ['xpto'].includes('xpt') which will return false.